src/fsipy/automatedPostprocessing/postprocessing_h5py/postprocessing_common_h5py.py

```diff
@@ -384,3 +343,305 @@ def create_transformed_matrix(input_path: Union[str, Path], output_folder: Union
+          <DataItem Dimensions="{num_nodes} {n_dim}" Format="HDF">{viz_type}.h5:/VisualisationVector/{idx}</DataItem>
+        </Attribute>
+      </Grid>
+'''
+
+        if idx == num_ts - 1:
+            break
+
+        lines += f'''\
+      <Grid>
+        <xi:include xpointer="xpointer(//Grid[@Name=&quot;TimeSeries_{viz_type}&quot;]/Grid[1]/*[self::Topology or self::Geometry])" />
+'''  # noqa
+
+    lines += '''\
+    </Grid>
+  </Domain>
+</Xdmf>
+'''
+
+    # Writing lines to file
+    xdmf_path = output_folder / f'{viz_type}.xdmf'
+
+    # Remove old file if it exists
+    if xdmf_path.exists():
+        logging.debug(f'--- Removing existing file at: {xdmf_path}')
+        xdmf_path.unlink()
+
+    with open(xdmf_path, 'w') as xdmf_file:
+        logging.debug(f'--- Writing XDMF file: {xdmf_path}')
+        xdmf_file.write(lines)
+
+
+def calculate_windowed_rms(signal_array: np.ndarray, window_size: int, window_type: str = "flat") -> np.ndarray:
+    """
+    Calculate the windowed root mean squared (RMS) amplitude of a signal.
+
+    Args:
+        signal_array (numpy.ndarray): Input signal.
+        window_size (int): Size of the window in number of timesteps.
+        window_type (str, optional): Type of window function to use. Default is "flat".
+            Supported window types: "flat", "tukey", "hann", "blackmanharris", "flattop".
+
+    Returns:
+        numpy.ndarray: Windowed RMS amplitudes.
+
+    References:
+         https://stackoverflow.com/questions/8245687/numpy-root-mean-squared-rms-smoothing-of-a-signal
+         https://stackoverflow.com/questions/47484899/moving-average-produces-array-of-different-length
+    """
+    # Calculate squared amplitudes of the signal
+    signal_squared = np.power(signal_array, 2)
+
+    # Define window functions for different types
+    window_functions = {
+        "flat": np.ones(window_size) / float(window_size),
+        "tukey": signal.windows.tukey(window_size) / float(window_size),
+        "hann": signal.windows.hann(window_size) / float(window_size),
+        "blackmanharris": signal.windows.blackmanharris(window_size) / float(window_size),
+        "flattop": signal.windows.flattop(window_size) / float(window_size),
+    }
+
+    # Select the appropriate window function based on the specified window type
+    window = window_functions.get(window_type, np.ones(window_size) / float(window_size))
+
+    # Calculate the RMS using convolution
+    RMS = np.sqrt(np.convolve(signal_squared, window, mode="valid"))
+
+    len_RMS = len(RMS)
+    len_signal_squared = len(signal_squared)
+    pad_length = int((len_signal_squared - len_RMS) / 2)
+
+    # Pad the RMS array to match the length of the original signal
+    RMS_padded = np.zeros(len_signal_squared)
+    for i in range(len_signal_squared):
+        if pad_length <= i < len_RMS + pad_length:
+            RMS_padded[i] = RMS[i - pad_length]
+
+    return RMS_padded
+
+
+def get_eig(T: np.ndarray) -> float:
```

Comment:
This function exists in turtle, so you can just import.

https://github.com/KVSlab/turtleFSI/blob/286fab226acb6fa35cf83cef95ef5cdd454026b7/turtleFSI/modules/common.py#L148